pkg/network/node/egressip.go

```diff
@@ -71,9 +75,13 @@ func newEgressIPWatcher(oc *ovsController, localIP string, masqueradeBit *int32)
 	return eip
 }
 
-func (eip *egressIPWatcher) Start(osdnInformers osdninformers.SharedInformerFactory, iptables *NodeIPTables) error {
+func (eip *egressIPWatcher) Start(osdnInformers osdninformers.SharedInformerFactory, kubeInformers informers.SharedInformerFactory, iptables *NodeIPTables) error {
 	eip.iptables = iptables
-	eip.tracker.Start(osdnInformers.Network().V1().HostSubnets(), osdnInformers.Network().V1().NetNamespaces())
+	if eip.cloudEgressIP {
+		eip.tracker.Start(osdnInformers.Network().V1().HostSubnets(), osdnInformers.Network().V1().NetNamespaces(), kubeInformers.Core().V1().Nodes())
```

Comment:
Does the node side of egressIP handling actually have to do anything special for cloud egress IPs? The cloud side deals with all the tracking and allocation, etc, and the node side is just reacting to NetNamespace.EgressIPs and HostSubnet.EgressIPs exactly like in the non-cloud case... It shouldn't need to know that the cloud egress IP controller is being used at all.

Reply:
> Does the node side of egressIP handling actually have to do anything special for cloud egress IPs?

No, but I guess my confusion stems from the fact that the validation of egress IP assignments is done in `common`, and this will now include cloud egress IP assignments. Ex: say a user performs manual allocations of egress IPs, if the nodes aren't able to validate the cloud egress IP assignment correctly (because they don't monitor nodes and hence have no clue what the capacity is), won't the nodes think that something is right when in fact sdn-controller knows that it's not? That would lead to nodes assigning the egress IP, while sdn-controller has dismissed it as invalid.

Reply:
All the validation stuff only happens in code paths that are only reached from `ReallocateEgressIPs`, which is only called by the master-side code. (So, really it shouldn't be in `common`, but as we discovered before, this code is a bit of a mess.)

Reply:
So as per the earlier comments here, this is not necessary; none of the cloud egress IP code will ever get run on the node side, so we _could_ just pass `false` to `NewEgressIPTracker` unconditionally. Arguably that would be kind of weird though... But at the same time, it's lame to set up a Node informer that we'll never use.

Oh well, I guess the current code is correct given the current `EgressIPTracker` API, and refactoring that is not currently on the table.

Reply:
OK, keeping it like this for now.